### PR TITLE
Week 1

### PR DESCRIPTION
**Pull request by seanclark84**
Simple hello world application which demonstrate autowiring and injecting using both annotation configuration as well as XML configuration.

In addition the project is a simple webapp which can be extended in future weeks. At the moment it simply displays a standard default welcome page. As well as this there is maven integration to handle the dependencies.

### CODE REVIEW

File: src/main/java/com/bankofspring/hello/SimpleHelloImpl.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  public void setArticle( String article ) {
+  	this.article = article;
+  }
+
+	/**
+   *
+   */
+  @Override
+  public void afterPropertiesSet() throws Exception {
+  	Assert.notNull( article, "An article is required" );
```
Comment (ramain75) [inline src/main/java/com/bankofspring/hello/SimpleHelloImpl.java:47]: :+1:

File: src/main/resources/hello/hello-applicationContext.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <bean id="helloArticle" class="java.lang.String">
+        <constructor-arg value="Sean"/>
+    </bean>
+
+    <!-- Get the autowired service -->
+    <context:component-scan base-package="com.bankofspring.hello"/>
+    
+    <!-- Or we can manually create one -->
+    <bean id="xmlSimpleHelloImpl" class="com.bankofspring.hello.SimpleHelloImpl">
+        <property name="article" ref="helloArticle"/>
```
Comment (ramain75) [inline src/main/resources/hello/hello-applicationContext.xml:19]: :+1:

File: src/test/java/com/bankofspring/hello/HelloTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	@Qualifier("annotationSimpleHelloImpl")
+	private HelloService annotationHelloService;
+	
+	@Autowired
+	@Qualifier("xmlSimpleHelloImpl")
+	private HelloService xmlHelloService;
+	
+	@Test
+	public void testAnnotationHelloService() {
+		assertThat( annotationHelloService.sayHello(), containsString( article ) );
```
Comment (ramain75) [inline src/test/java/com/bankofspring/hello/HelloTest.java:36]: I really like the way you've done this, ensuring that you autowire article. 

What message do you get if the assert fails compared with if you used the `assertThat( "failure message", annotationHelloService.sayHello(), containsString( article ) );`?

Comment (seanclark84) [inline src/test/java/com/bankofspring/hello/HelloTest.java:36]: The messages are very useful and also it makes it really easy to create your own matcher:

```
Expected: a string containing "bob"
     but: was "Hello Sean"
```

or

```
Expected: "bob"
     but: was "Hello Sean"
```

**Pull request closed without merging**